### PR TITLE
Fix search of repository root in Windows

### PR DESCRIPTION
**Pull request by jsoriano**
When looking for the repository root in Windows, if the command hasn't been called inside a git repository, the search will loop forever, hanging the process. Same thing happens when looking for package root.

Finalization condition is based on checking all parent directories till root, but root was checked with "/", that will never be the case in Windows.

Changed to take into account the volume name and the OS-specific path separator.

Add step to buildkite pipeline to run unit tests on Windows.

Fixes https://github.com/elastic/elastic-package/issues/1100.

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -24,6 +24,13 @@ steps:
       cpu: "8"
       memory: "4G"
 
+  - label: ":go: :windows: Run unit tests"
+    key: unit-tests-windows
+    command: ".buildkite/scripts/unit_tests_windows.ps1"
+    agents:
+      provider: "gcp"
+      image: "family/ci-windows-2022"
```
Comment (mrodm) [inline .buildkite/pipeline.yml:32]: Great to have this running also in Windows :+1: 

I would also add this step as a dependency of the step related to integration tests:

```yaml
  - label: ":pipeline: Trigger Integration tests"
    command: ".buildkite/pipeline.trigger.integration.tests.sh | buildkite-agent pipeline upload"
    depends_on:
      - step: check-static
        allow_failure: false
      - step: unit-tests-linux
        allow_failure: false
      - step: unit-tests-windows
        allow_failure: false
```
Doing so, if the unit tests in windows fail, integration tests won't be triggered.

**Review by mrodm — COMMENTED** [review]
Thanks for adding the step for Windows :+1:

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (jsoriano) [reply .buildkite/pipeline.yml:32]: Added, thanks :+1:

**Review by mrodm — APPROVED** [review]

**Pull request merged**